File: ios/Classes/Common/LingYeBaseRegister.swift
import Foundation
import Flutter

class LingYeBaseRegister: NSObject {

    var registrar: FlutterPluginRegistrar?
    var channel: FlutterMethodChannel?
    static let shared: LingYeBaseRegister = LingYeBaseRegister()
    override init() {
        super.init()
    }
    
    func registrar(register: FlutterPluginRegistrar) -> Void {
        self.registrar = register
        
        channel = FlutterMethodChannel(name: "flutter_cjadsdk_plugin", binaryMessenger: register.messenger())
        channel?.setMethodCallHandler { methodCall, result in
            print("事件" + methodCall.method)
            switch methodCall.method {
            case LingYeAdSdkMethodNames.initSDKFun:
                LingYeSetupHandler.shared.handleMethodCall(methodCall, result: result)
            case LingYeAdSdkMethodNames.loadAndShowSplashAd:
                 LingYeSplashHandler.shared.handleMethodCall(methodCall, result: result)
            case LingYeAdSdkMethodNames.loadAndShowInterstitialAd:
                LingYeInterstitialHandler.shared.handleMethodCall(methodCall, result: result)
            case LingYeAdSdkMethodNames.loadAndShowRewardVideoAd:
                LingYeRewardVideoHandler.shared.handleMethodCall(methodCall, result: result)
            case LingYeAdSdkMethodNames.loadNativeAd:
                LingYeNativeHandler.shared.handleMethodCall(methodCall, result: result)
            case LingYeAdSdkMethodNames.loadIconAd:
                LingYeIconHandler.shared.handleMethodCall(methodCall, result: result)
            default:
                result(FlutterMethodNotImplemented)
            }
        }
    }
    
    func sendToFlutter(method: String, arguments: Any) {
        DispatchQueue.main.async {
            self.channel?.invokeMethod(method, arguments: arguments)
        }
    }
    
}

File: android/src/main/kotlin/com/example/flutter_cjadsdk_plugin/common/LingYeBaseRegister.kt
import android.util.Log
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler

class LingYeBaseRegister private constructor() : MethodCallHandler {

    private var channel: MethodChannel? = null

    companion object {
        private var sInstance: LingYeBaseRegister? = null
        fun getInstance(): LingYeBaseRegister {
            return sInstance ?: synchronized(this) {
                sInstance ?: LingYeBaseRegister().also { sInstance = it }
            }
        }
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        Log.e("cj-log", "开始调用事件： " + call.method)
        when(call.method){
            LingYeAdSdkMethodNames.initSDKFun -> {
                LingYeSetupHandler.getInstance().handleMethodCall(call, result)
            }
            LingYeAdSdkMethodNames.loadAndShowSplashAd -> {
                LingYeSplashHandler.getInstance().handleMethodCall(call, result)
            }
            LingYeAdSdkMethodNames.loadAndShowInterstitialAd -> {
                LingYeInterstitialHandler.getInstance().handleMethodCall(call, result)
            }
            LingYeAdSdkMethodNames.loadAndShowRewardVideoAd -> {
                LingYeRewardHandler.getInstance().handleMethodCall(call, result)
            }
            LingYeAdSdkMethodNames.loadIconAd -> {
                LingYeIconHandler.getInstance().handleMethodCall(call, result)
            }
        }
    }

    fun init(binaryMessenger: BinaryMessenger) {
        if (channel == null) {
            channel = MethodChannel(binaryMessenger, "flutter_cjadsdk_plugin")
            channel?.setMethodCallHandler(this)
        }
    }

    fun sendMessageToFlutter(method: String, args: Any?) {
        LingYeTools.runOnThread {
            channel?.invokeMethod(method, args)
        }
    }


}